src/System.IO.FileSystem/src/System/IO/FileSystemInfo.Unix.cs

```diff
@@ -202,15 +204,17 @@ DateTimeOffset IFileSystemObject.CreationTime
                 if (!_exists)
                     return DateTimeOffset.FromFileTime(0);
 
-                long rawTime = (_fileStatus.Flags & Interop.Sys.FileStatusFlags.HasBirthTime) != 0 ?
+                Interop.Sys.TimeSpec rawTime = (_fileStatus.Flags & Interop.Sys.FileStatusFlags.HasBirthTime) != 0 ?
                     _fileStatus.BirthTime :
-                    Math.Min(_fileStatus.CTime, _fileStatus.MTime); // fall back to the oldest time we have in between change and modify time
-                return DateTimeOffset.FromUnixTimeSeconds(rawTime).ToLocalTime();
+                     // fall back to the oldest time we have in between change and modify time
+                    _fileStatus.CTime < _fileStatus.MTime ? _fileStatus.CTime : _fileStatus.MTime;
+
+                return DateTimeOffset.FromUnixTimeSeconds(rawTime.Seconds).ToLocalTime().AddTicks(rawTime.Nanoseconds / NanosecondsPerTick);
             }
```

Comment:
Helper method for the `DateTimeOffset.FromUnixTimeSeconds(_fileStatus.MTime.Seconds).ToLocalTime().AddTicks(_fileStatus.MTime.Nanoseconds / NanosecondsPerTick)` sequence?